### DISCUSSION TITLE
Server instances

### DISCUSSION THREAD

**Discussion by MrPRAINGLE** [discordjs/discord.js#6952] [Q&A] [answered]
Hey ! I recentlly discovered the trick with the .json files for multiples bot instances, but I have no informations how it works. Can you help me please ? Thanks ! (I did not find any tutorials in the discord.js guide, YouTube or any forum. If i is already answered, do not waste time to explain me and post directly the link, thanks !)

**Comment by vvito7** [comment]
what do you mean by `bot instances`?

**Reply by jerryxfu** [reply]
And also what ".json" file are you talking about?

**Reply by vvito7** [reply]
i'm not sure, but i guess it's the `config.json`, that keeps the sensitives values

**Reply by MrPRAINGLE** [reply]
For an example, when the bot joins a new guild, a ``.json`` file is created, with things such as prefix, moderation settings... And using that allows to have a different prefix for different guilds, while the bot keeps bing the same (that's what I wall "bot instances", and it is ilke a ``config.json`` but without token and it is a different ``config.json`` for each guild)

**Reply by Chere3** [reply]
I mean, you want to make a multi-prefix bot that does a `.json` per server?

**Reply by vvito7** [reply]
@Chere3 i guess they just want to know how to use instances from a `.json`

**Reply by Chere3** [reply]
> @Chere3 i guess they just want to know how to use instances from a `.json`

Literally like a database, no?

**Reply by vvito7** [reply]
yes, i guess
like the bot token, you know?

**Comment by vvito7** [comment]
ok so if i understood, you only want to know how to use the `.json` for instances in your code
this is really simple:

in the `.json` file, write the intance that you want to use:

![image](https://user-images.githubusercontent.com/79790819/141088372-e377f49a-0347-4589-a337-f6e708c4794c.png)

and then to use in your code, you can import it like this (you just need to know the path to `.json` file):

![image](https://user-images.githubusercontent.com/79790819/141088592-32c7a761-6446-4bf7-84ba-aa839470c867.png)

**Reply by MrPRAINGLE** [reply]
So I would need to do something like ``Client.on("messageCreate", message => if (message.guild.id === "an actual ID") {const { something, anotherthing } = require("./configforguild#42.json"})`` ?

**Reply by LukeHankey** [reply]
It's better here if you actually hook up a database and create a new table/schema/construct for each new guild since json files, when used as a database with frequent read/write operations are prone to corruption.

The website has a simple guide for the [sequelize database (ORM)](https://discordjs.guide/sequelize/).

**Reply by vvito7** [reply]
> So I would need to do something like `Client.on("messageCreate", message => if (message.guild.id === "an actual ID") {const { something, anotherthing } = require("./configforguild#42.json"})` ?

no, you just import it on the top of the code and then uses `something` like that: 
```
const { something, anotherthing } = require("./configforguild#42.json")

Client.on("messageCreate", message => if (message.guild.id === "an actual ID") { something }
```

**Reply by MrPRAINGLE** [reply]
Thanks for your support, @LukeHankey it's basically what I searched ! I just did not see it on the site... Thanks !!

**Comment by callumok2004** [comment, marked as answer]
Please use a database for guild configuration, do not use JSON files as a database!

**Reply by MrPRAINGLE** [reply]
Ok thanks !
